pkg/gadgets/dns/tracer/tracer.go

```diff
@@ -215,6 +215,7 @@ func (t *Tracer) listen(rd *perf.Reader, f func(name, pktType string)) {
 				return
 			}
 			log.Errorf("Error while reading from perf event reader: %s", err)
+			return
```

Comment:
I'm wondering if we should only return on some non-recoverable errors... But it seems complicated to understand which ones are those...

Reply:
Uhm, I think that if rd.Read() returns an error, then we can't use the record variable anymore and thus we should always return.

Reply:
But we could do a `continue` to try again if the error is recoverable.